cmd/lotus-chainwatch/processor/miner.go

```diff
@@ -451,6 +481,88 @@ func (p *Processor) storeMinersSectorHeads(miners []minerActorInfo) error {
 	return tx.Commit()
 }
 
+func (p *Processor) storeMinersPreCommitState(ctx context.Context, miners []minerActorInfo) error {
+	start := time.Now()
+	defer func() {
+		log.Infow("Stored Miners Precommit State", "duration", time.Since(start).String())
+	}()
+
+	precommitTx, err := p.db.Begin()
+	if err != nil {
+		return err
+	}
+
+	eventTx, err := p.db.Begin()
+	if err != nil {
+		return err
+	}
+
+	if _, err := eventTx.Exec(`create temp table mse (like miner_sector_events excluding constraints) on commit drop;`); err != nil {
+		return xerrors.Errorf("prep temp: %w", err)
+	}
+
+	eventStmt, err := eventTx.Prepare(`copy mse (sector_id, event, miner_id, state_root) from STDIN `)
+	if err != nil {
+		return err
+	}
+
+	if _, err := precommitTx.Exec(`create temp table mp (like miner_precommits excluding constraints) on commit drop;`); err != nil {
+		return xerrors.Errorf("prep temp: %w", err)
+	}
+
+	precommitStmt, err := precommitTx.Prepare(`copy mp (miner_id, sector_id, precommit_deposit, precommit_epoch) from STDIN`)
+	if err != nil {
+		return err
+	}
+
+	for _, m := range miners {
+		m := m
+		pcMap, err := adt.AsMap(cw_util.NewAPIIpldStore(ctx, p.node), m.state.PreCommittedSectors)
+		if err != nil {
+			return err
+		}
+		precommit := new(miner.SectorPreCommitOnChainInfo)
+		if err := pcMap.ForEach(precommit, func(key string) error {
+			if _, err := precommitStmt.Exec(
+				m.common.addr.String(),
+				precommit.Info.SectorNumber,
+				precommit.PreCommitDeposit.String(),
+				precommit.PreCommitEpoch,
+			); err != nil {
+				return err
```

Comment:
Thinking about this return (and others) ...I'm not sure we want to? Maybe log an error?